index.js

```diff
@@ -0,0 +1,168 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const append = require('vary').append
+
+function fastifyCors (fastify, opts, next) {
+  const {
+    origin,
+    credentials,
+    exposedHeaders,
+    allowedHeaders,
+    methods,
+    maxAge,
+    preflightContinue,
+    optionsSuccessStatus
+  } = Object.assign({
+    origin: '*',
+    methods: 'GET,HEAD,PUT,PATCH,POST,DELETE',
+    preflightContinue: false,
+    optionsSuccessStatus: 204,
+    credentials: false,
+    exposedHeaders: null,
+    allowedHeaders: null,
+    maxAge: null
+  }, opts)
+
+  const isOriginFalsy = !origin
+  const isOriginString = typeof origin === 'string'
+  const isOriginFunction = typeof origin === 'function'
+
+  fastify.options('*', (req, reply) => reply.send())
```

Comment:
I think this should be behind an option/and configurable. Somebody might want to specify this only for certain endpoints.

Reply:
How would you name this option?

Reply:
`preflight`, it can be a `String` or an `Array`. Default `'*'` (what you have right now).
If `''` or `null`, nothing happens. If an array, all elements are passed to options.

Reply:
For now, I've added the option to disable `preflight` entirely. We should do some work in core to help people understand if you are inside a 404 context.